packages/ai-ide/src/browser/ai-configuration/agent-configuration-widget.tsx

```diff
@@ -88,97 +89,140 @@ export class AIAgentConfigurationWidget extends ReactWidget {
+                this.languageModelAliases = this.languageModelAliasRegistry.getAliases();
+                this.languageModels = models;
+                this.update();
+            }),
+            this.promptService.onPromptsChange(() => this.updateParsedPromptParts()),
+            this.promptFragmentCustomizationService.onDidChangePromptFragmentCustomization(() => {
+                this.updateParsedPromptParts();
+            }),
+            this.aiSettingsService.onDidChange(() => {
+                this.updateParsedPromptParts();
+            }),
+            this.aiConfigurationSelectionService.onDidAgentChange(() => {
+                this.selectedItem = this.aiConfigurationSelectionService.getActiveAgent();
+                this.updateParsedPromptParts();
+            }),
+            this.agentService.onDidChangeAgents(async () => {
+                await this.loadItems();
+                this.update();
+            })
+        ]);
 
-        // Only call updateParsedPromptParts() as it already calls this.update() internally
-        this.toDispose.push(this.aiSettingsService.onDidChange(() => {
-            this.updateParsedPromptParts();
-        }));
+        this.updateParsedPromptParts();
+    }
 
-        // Only call updateParsedPromptParts() as it already calls this.update() internally
-        this.toDispose.push(this.aiConfigurationSelectionService.onDidAgentChange(() => {
-            this.updateParsedPromptParts();
-        }));
+    protected async loadItems(): Promise<void> {
+        this.items = this.agentService.getAllAgents();
+        const activeAgent = this.aiConfigurationSelectionService.getActiveAgent();
+        if (activeAgent) {
+            this.selectedItem = activeAgent;
+        } else if (this.items.length > 0 && !this.selectedItem) {
+            this.selectedItem = this.items[0];
+            this.aiConfigurationSelectionService.setActiveAgent(this.items[0]);
+        }
+    }
+
+    protected getItemId(agent: Agent): string {
+        return agent.id;
+    }
+
+    protected getItemLabel(agent: Agent): string {
+        return agent.name;
+    }
 
-        this.agentService.onDidChangeAgents(() => this.update());
+    protected override getEmptySelectionMessage(): string {
+        return nls.localize('theia/ai/core/agentConfiguration/selectAgentMessage', 'Please select an Agent first!');
+    }
+
+    protected override handleItemSelect = (agent: Agent): void => {
+        this.selectedItem = agent;
+        this.aiConfigurationSelectionService.setActiveAgent(agent);
         this.updateParsedPromptParts();
+    };
+
+    protected override renderItemPrefix(agent: Agent): React.ReactNode {
+        const enabled = this.agentService.isEnabled(agent.id);
+        return (
+            <span className={`agent-status-indicator ${enabled ? 'agent-enabled' : 'agent-disabled'}`} title={enabled ? 'Enabled' : 'Disabled'}>
+                ●
+            </span>
```

Comment:
When an item is selected or for some color themes it is hard to see the difference, so i would suggest to use a different icon additionally if the it is disabled, e.g. something like:
```
<span className={`agent-status-indicator ${enabled ? `agent-enabled ${codicon('circle-filled')}` : `agent-disabled ${codicon('circle')}`}`} title={enabled ? 'Enabled' : 'Disabled'}>
```